aws/sdk/s3-benchmark/benchmark/src/multipart_put.rs

```diff
@@ -3,26 +3,60 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+use crate::put_test::PutBenchmark;
 use crate::{Args, BoxError, BENCH_KEY};
+use async_trait::async_trait;
+use aws_config::SdkConfig;
 use aws_sdk_s3 as s3;
+use aws_sdk_s3::Client;
 use aws_smithy_http::byte_stream::ByteStream;
-use aws_smithy_http::byte_stream::Length;
 use s3::types::CompletedMultipartUpload;
 use s3::types::CompletedPart;
+use std::io::SeekFrom;
 use std::path::Path;
 use std::path::PathBuf;
 use std::sync::Arc;
-use tokio::sync::Semaphore;
+use std::time::{Duration, SystemTime};
+use tokio::fs::File;
+use tokio::io::{AsyncReadExt, AsyncSeekExt};
+use tokio::sync::{OwnedSemaphorePermit, Semaphore};
+use tokio::time::timeout;
+
+pub(crate) struct PutObjectMultipart;
+
+#[async_trait]
+impl PutBenchmark for PutObjectMultipart {
+    type Setup = Vec<Client>;
+
+    async fn prepare(&self, conf: &SdkConfig) -> Self::Setup {
+        let clients = (0..32).map(|_| Client::new(&conf)).collect::<Vec<_>>();
+        for client in &clients {
+            let _ = client.list_buckets().send().await;
+        }
```

Comment:
What's the purpose of listing buckets with no arguments in this preparation step?

Reply:
preload credentials